gemma.cc

```diff
@@ -529,7 +529,7 @@ void GenerateImpl(GemmaImpl<TConfig>& gemma, const InferenceArgs& args,
 
   // Prefill stops before prompt.size() - 1 since the last prompt token is the
   // first input token for generation.
-  while (pos_offset < prompt.size() - 1) {
+  while (pos_offset < prompt.size() - 1 && token != EOS_ID) {
```

Comment:
Is there a reason to override the token value vs. track the stream_token return state with its own bool variable? Keeping the two forms of state seems easier to reason about rather than having to track "Is token EOS_ID because that's what was generated or because stream_token was false..."

Reply:
You are right. I want to change as little code as possible. But it's not meaningful.
I will think more again.